eclair-core/src/main/scala/fr/acinq/eclair/channel/Helpers.scala

```diff
@@ -43,6 +43,37 @@ import scala.util.{Failure, Success, Try}
 
 object Helpers {
 
+  /**
+   * When we receive the preimage for an HTLC and want to fulfill it but the upstream peer stops responding, we want to
+   * avoid letting its HTLC-timeout transaction become enforceable on-chain (otherwise there is a race condition between
+   * our HTLC-success and their HTLC-timeout).
+   * We will close the channel when the HTLC-timeout will happen in less than N blocks, where we scale N linearly depending
+   * on the total amount at risk (there may be multiple HTLCs in that state).
+   *
+   * @param min           if incoming HTLCs expire in less than `min`, we close the channel regardless of the amount at risk.
+   * @param max           incoming HTLCs that expire in more than `max` are ignored.
+   * @param deltaPerBlock linear scaling factor for the safety window.
+   */
+  case class FulfillSafetyBeforeTimeout(min: CltvExpiryDelta, max: CltvExpiryDelta, deltaPerBlock: MilliSatoshi) {
```

Comment:
I would have put this somewhere else, maybe right below the `aboveReserve` method.